Lib/fontTools/cffLib/cff2_merge_funcs.py

```diff
@@ -0,0 +1,269 @@
+		error_msg = ["For the Private Dict key ()".format(key)]
+		error_msg.append("the default font value list:")
+		error_msg.append("\t{}".format(value))
+		error_msg.append(
+					"had a different number of values than"
+					"a region font:")
+		for value in values:
+			error_msg.append("\t{}".format(value))
+		error_msg = os.linesep.join(error_msg)
+
+
+def conv_to_int(num):
+	if round(num) == num:
+		return int(num)
+	else:
+		return num
+
+
+def merge_PrivateDicts(topDict, region_top_dicts, num_masters, var_model):
+	if hasattr(region_top_dicts[0], 'FDArray'):
+		regionFDArrays = [fdTopDict.FDArray for fdTopDict in region_top_dicts]
+	else:
+		regionFDArrays = [[fdTopDict] for fdTopDict in region_top_dicts]
+	for fd_index, font_dict in enumerate(topDict.FDArray):
+		private_dict = font_dict.Private
+		pds = [private_dict] + [
+			regionFDArray[fd_index].Private for regionFDArray in regionFDArrays
+			]
+		for key, value in private_dict.rawDict.items():
+			if isinstance(value, list):
+				try:
+					values = [pd.rawDict[key] for pd in pds]
+				except KeyError:
+					del private_dict.rawDict[key]
+					print(
+						b"Warning: {key} in default font Private dict is "
+						b"missing from another font, and was "
+						b"discarded.".format(key=key))
+					continue
+				try:
+					values = zip(*values)
+				except IndexError:
+					raise MergeDictError(key, value, values)
+				"""
+				Row 0 contains the first  value from each master.
+				Convert each row from absolute values to relative
+				values from the previous row.
+				e.g for three masters,	a list of values was:
+				master 0 OtherBlues = [-217,-205]
+				master 1 OtherBlues = [-234,-222]
+				master 1 OtherBlues = [-188,-176]
+				The call to zip() converts this to:
+				[(-217, -234, -188), (-205, -222, -176)]
+				and is converted finally to:
+				OtherBlues = [[-217, 17.0, 46.0], [-205, 0.0, 0.0]]
+				"""
+				dataList = []
+				prev_val_list = [0] * num_masters
+				any_points_differ = False
+				for val_list in values:
+					rel_list = [(val - prev_val_list[i]) for (
+							i, val) in enumerate(val_list)]
+					if (not any_points_differ) and pointsDiffer(rel_list):
+						any_points_differ = True
+					prev_val_list = val_list
+					deltas = var_model.getDeltas(rel_list)
+					# Convert numbers with no decimal part to an int.
+					deltas = [conv_to_int(delta) for delta in deltas]
+					# For PrivateDict BlueValues, the default font
+					# values are absolute, not relative to the prior value.
+					deltas[0] = val_list[0]
+					dataList.append(deltas)
+				# If there are no blend values,then
+				# we can collapse the blend lists.
+				if not any_points_differ:
+					dataList = [data[0] for data in dataList]
+			else:
+				values = [pd.rawDict[key] for pd in pds]
+				if pointsDiffer(values):
+					dataList = var_model.getDeltas(values)
+				else:
+					dataList = values[0]
+			private_dict.rawDict[key] = dataList
+
+
+class MergeCharError(TypeError):
```

Comment:
I don't understand the need for this class. Seems like an overcomplicated way to collect and raise errors.

Reply:
I'll leave it to others to weigh in. The solution seems over-engineered to me.